### PR TITLE
feat: add series and expr `clip` method

### LINKED ISSUE #700: feat: Add clip expression to the api
### We would like to learn about your use case. For example, if this feature is needed  to adopt Narwhals in an open source project, could you please enter the link to it below?

_No response_

### Please describe the purpose of the new feature or describe the problem to solve.

It would be a nice QOL and likely a speed enhancement to have a `clip` expression in the api

Since it is a simple mathematical function, it seems like `polars` and `pandas` largely agree on the api, looking something like - `clip(lower_limit, upper_limit, *, **kwargs)`

### Suggest a solution if possible.

Adding a `polars` looking expression `clip(lower_bound, upper_bound)` would solve this issue

### If you have tried alternatives, please describe them below.

_No response_

### Additional information that may help us understand your needs.

_No response_

### PR DESCRIPTION
**Pull request by aivanoved**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #700 
- Closes #700

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: tests/expr_and_series/clip_test.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import pytest
+
+import narwhals.stable.v1 as nw
+from tests.utils import compare_dicts
+
+
+def test_clip(request: Any, constructor: Any) -> None:
+    if "pyarrow_table" in str(constructor):
+        request.applymarker(pytest.mark.xfail)
```
Comment (FBruzzesi) [inline tests/expr_and_series/clip_test.py:11]: Also dask should be marked to fail:

```suggestion
    if "pyarrow_table" in str(constructor) or "dask" in str(constructor):
        request.applymarker(pytest.mark.xfail)
```

File: narwhals/expr.py [hunk 2]
```diff
@@ -1765,6 +1765,111 @@ def gather_every(self: Self, n: int, offset: int = 0) -> Self:
...
+        self,
+        lower_bound: IntoExpr | Any | None = None,
+        upper_bound: IntoExpr | Any | None = None,
+    ) -> Self:
+        r"""
+        Clip values in the Series.
+        Arguments:
+            lower_bound: Lower bound value.
+            upper_bound: Upper bound value.
+        Examples:
```
Comment (FBruzzesi) [inline narwhals/expr.py:1780]: In this example we should exemplify the behavior on an expr, not on a series, namely something like `df.select(nw.col("a").clip(2))`

Comment (FBruzzesi) [inline narwhals/expr.py:1780]: Could you make sure that docs render properly without newlines across keywords?

File: narwhals/series.py [hunk 3]
```diff
@@ -544,6 +544,105 @@ def std(self, *, ddof: int = 1) -> Any:
...
 
+    def clip(
+        self, lower_bound: Any | None = None, upper_bound: Any | None = None
+    ) -> Self:
+        r"""
+        Clip values in the Series.
+        Arguments:
+            lower_bound: Lower bound value.
+            upper_bound: Upper bound value.
+        Examples:
```
Comment (FBruzzesi) [inline narwhals/series.py:555]: As above, please make sure that rendering is ok

**Review by FBruzzesi — COMMENTED** [review]
Thanks for the PR, it is almost there, just a few details missing or to confirm 🙌🏼

File: narwhals/expr.py (hunk 2 above)
Comment (aivanoved) [reply narwhals/expr.py:1780]: thank you for the catch, I had forgotten to update after copy pasting

Comment (aivanoved) [reply narwhals/expr.py:1780]: Yes, thank you for catching this as well

Comment (aivanoved) [reply narwhals/expr.py]: should be fixed

**Review by FBruzzesi — APPROVED** [review]
Thanks for fixing 🙌🏼

JackSparrowCatKittyGIF

**Pull request merged**